integration_tests/README.md

```diff
@@ -60,6 +60,33 @@ The python tests run with pytest and the script honors pytest parameters. Some h
 - `-r fExXs` Show extra test summary info as specified by chars: (f)ailed, (E)rror, (x)failed, (X)passed, (s)kipped
 - For other options and more details please visit [pytest-usage](https://docs.pytest.org/en/stable/usage.html) or type `pytest --help`
 
+### Spark execution mode
+
+Spark Applications (pytest in this case) can be run against different cluster backends
+specified by the configuration `spark.master`. It can be provided by various means such
+as via `--master` argument of `spark-submit`.
+
+By default, the [local mode](
+https://github.com/apache/spark/blob/1a042cc414c0c720535798b9a1197fe8885d6f6e/core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala#L214
+) is used to run the Driver and Executors in the same JVM. Albeit convenient, this mode sometimes
+masks problems occurring in fully distributed production deployments. These are often bugs related
+to object serialization and hash code implementation.
+
+Thus, Apache Spark provides another lightweight way to test applications in the pseudo-distributed
+[local-cluster[numWorkers,coresPerWorker,memoryPerWorker]](
+https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L3025
```

Comment:
nit: would be nice to point this to a specific commit blob like [latest commit](https://github.com/apache/spark/blob/d424771ec2c061dc5fa237b192abcf6aaece2616/core/src/main/scala/org/apache/spark/SparkContext.scala#L3025) in case the line number changed